### PR TITLE
Zqs 1066/polish vqa guide

### PR DESCRIPTION
**Pull request by mstechly**
## Description

Update of the VQA guide and related pieces of docs.

## Please verify that you have completed the following steps

- [X] I have self-reviewed my code.
- [X] I have updated documentation.

### CODE REVIEW

File: docs/guides/vqa_guide/ansatzes.rst [hunk 1]
```diff
@@ -50,4 +49,11 @@ Finally, we'll define the ``_generate_circuit`` function which substitutes param
     :start-at: def _generate_circuit(
     :end-at: return self._circuit.bind(symbols_map)
 
-And that's it! Once the ansatz is defined we can easily subtitute in into an ``EstimationTaskFactory``.
\ No newline at end of file
+And that's it! Once the ansatz is defined we can easily subtitute in into an ``EstimationTaskFactory``.
+
+
+Another important thing to know about our implementation of ansatzes is that it can support symbolic parameters. Conceptually, ansatzes yield themselves nicely be expressed using symbols rather. However, it might come at the price of performance, so we recommend implementing your ansatz using symbolic circuits in two cases:
```
Comment (Boniface316) [inline docs/guides/vqa_guide/ansatzes.rst:55]: Instead of saying "important thing to know", reword it to "important concept to understand"

File: docs/guides/vqa_guide/ansatzes.rst [hunk 2]
```diff
@@ -50,4 +49,11 @@ Finally, we'll define the ``_generate_circuit`` function which substitutes param
...
+
+- if generating circuit is time-consuming. This is the case e.g. when you need to perform time evolution of some Hamiltonian to build the circuit.
```
Comment (Boniface316) [inline docs/guides/vqa_guide/ansatzes.rst:57]: Not clear whats mentioned here.

File: docs/guides/vqa_guide/estimators.rst [hunk 3]
```diff
@@ -4,46 +4,162 @@
...
+
+1. Return one ExpectationValue for each EstimationTask.
+2. The order in which ExpectationValues are returned should match the order in which EstimationTasks were provided.
+3. Number of entries in each ExpectationValue is not restricted - if the operator consists of multiple terms, estimator might be contain values for each term.
+4. Output ExpectationValue should include coefficients of the terms/operators.
+5. ``estimation_tasks`` can include tasks where operator consists of a constant term or contains a constant term. The implemented method should include the contributions of such constant terms in the return value.
+
+I hope points 1-3 are not controversial for you. Points 4-5 define a conventions that we have decided to follow - we know that in some circles people like using other conventions, which might be confusing. So we want to be clear about the conventions that we follow.
```
Comment (Boniface316) [inline docs/guides/vqa_guide/estimators.rst:92]: Using the term "I" allowed in the documentation? If no, we can reword it as "An assumption is made here that points 1-3 are not controversial for the reader."

File: docs/guides/vqa_guide/estimators.rst [hunk 4]
```diff
@@ -4,46 +4,162 @@
...
+Some other tools:
+
+- :func:`evaluate_estimation_circuits <orquestra.quantum.estimation.evaluate_estimation_circuits>` - substitutes symbols with numerical parameters for all the circuits in the estimation tasks 
+- :func:`split_estimation_tasks_to_measure <orquestra.quantum.estimation.split_estimation_tasks_to_measure>` - this function splits a given list of EstimationTask into two: one that contains EstimationTasks that should be measured, and one that contains EstimationTasks with constants or with 0 shots.
+- :func:`evaluate_non_measured_estimation_tasks <orquestra.quantum.estimation.evaluate_non_measured_estimation_tasks>` - this returns the expectation value for the tasks which don't need to be measured
+
+Example
+=======
+
+Let's see how this plays out in an actual example. For the sake of example, let's say we want to run VQE using the following techniques:
```
Comment (Boniface316) [inline docs/guides/vqa_guide/estimators.rst:146]: Remove "For the sake of example" and say "For example"

File: docs/guides/vqa_guide/structure.rst [hunk 5]
```diff
@@ -47,11 +45,26 @@ We'll go through the steps as they were outlined above:
 
 1. The ``Ansatz``, ``Backend`` and ``Optimizer`` can be defined independently of one another.
 2. As explained in :ref:`estimators guide <estimators>`, the ``EstimationTaskFactory`` tells the quantum computer how to construct a circuit. It therefore will always take an ``Ansatz`` as an argument.
-3. As explained briefly above, VQA uses a quantum computer to estimate the cost function. So it is sensible that to create our cost function we would need the method of costructing circuits for our cost function (``EstimationTask``) and the quantum computer to run them on (``QuantumBackend``). See the :ref:`cost function <cost_function>` guide for details.
+3. As explained briefly above, VQA uses a quantum computer to estimate the value of the cost function. So to create our cost function we would need the method of costructing circuits for our cost function (``estimation_task_factory``) and the quantum computer to run them on (``QuantumBackend``). See the :ref:`cost function <cost_function_guide>` guide for details.
 4. Now that we have the , we minimize if using the optimizer. This step generally takes the longest to run.
```
Comment (Boniface316) [inline docs/guides/vqa_guide/structure.rst:49]: 4. is incomplete

File: docs/guides/vqa_guide/estimators.rst (hunk 3 above)
Comment (mstechly) [reply docs/guides/vqa_guide/estimators.rst:92]: I just changed it to "we", as I'm definitely using "we" across the docs.

File: docs/guides/vqa_guide/ansatzes.rst (hunk 1 above)
Comment (mstechly) [reply docs/guides/vqa_guide/ansatzes.rst]: Fixed in https://github.com/zapatacomputing/orquestra-core/pull/60/commits/0b3d2fd9b5438795d3280b6ecad3885360af1ba2

File: docs/guides/vqa_guide/ansatzes.rst (hunk 2 above)
Comment (mstechly) [reply docs/guides/vqa_guide/ansatzes.rst]: Fixed in https://github.com/zapatacomputing/orquestra-core/pull/60/commits/0b3d2fd9b5438795d3280b6ecad3885360af1ba2

File: docs/guides/vqa_guide/estimators.rst (hunk 4 above)
Comment (mstechly) [reply docs/guides/vqa_guide/estimators.rst]: Fixed in https://github.com/zapatacomputing/orquestra-core/pull/60/commits/0b3d2fd9b5438795d3280b6ecad3885360af1ba2

File: docs/guides/vqa_guide/structure.rst (hunk 5 above)
Comment (mstechly) [reply docs/guides/vqa_guide/structure.rst]: Fixed in https://github.com/zapatacomputing/orquestra-core/pull/60/commits/0b3d2fd9b5438795d3280b6ecad3885360af1ba2

**Review by Boniface316 — APPROVED** [review]
